packages/react/src/select/trigger/useSelectTrigger.ts

```diff
@@ -44,20 +45,24 @@ export function useSelectTrigger(
 
   const handleRef = useForkRef<HTMLElement>(buttonRef, setTriggerElement);
 
+  const timeout1 = useTimeout();
+  const timeout2 = useTimeout();
+
   React.useEffect(() => {
     if (open) {
-      // mousedown -> mouseup on selected item should not select within 400ms.
-      const timeoutId1 = window.setTimeout(() => {
-        selectionRef.current.allowSelectedMouseUp = true;
-      }, 400);
       // mousedown -> move to unselected item -> mouseup should not select within 200ms.
-      const timeoutId2 = window.setTimeout(() => {
+      timeout1.start(200, () => {
         selectionRef.current.allowUnselectedMouseUp = true;
-      }, 200);
+
+        // mousedown -> mouseup on selected item should not select within 400ms.
+        timeout2.start(200, () => {
+          selectionRef.current.allowSelectedMouseUp = true;
+        });
+      });
```

Comment:
I don't think these should be nested as before?

Reply:
I'm pretty sure it's equivalent to the unnested version. It felt a bit more efficient perf-wise to nest it, but reading the context again (select open event) I think we can ignore the perf aspect for this case. I've reverted this change.